### PR TITLE
SWP-88: Implements latest version of SDK in Bridgecontroller

### PR DESCRIPTION
**Pull request by m038**
Adds support for authentication and new client structure.

### CODE REVIEW

File: Controller/BridgeController.php [hunk 1]
```diff
@@ -83,29 +86,36 @@ public function indexAction(Request $request, $endpoint, $objectId = null)
...
-
-        return new GuzzleClient($bridgeConfig);
+        $authentication = new OAuthPasswordAuthentication(new GuzzleClient());
+        $authentication
+            ->setClientId($this->container->getParameter('swp_bridge.auth.client_id'))
+            ->setUsername($this->container->getParameter('swp_bridge.auth.username'))
+            ->setPassword($this->container->getParameter('swp_bridge.auth.password'));
+        $apiClient = new GuzzleApiClient(new GuzzleClient(), $authentication);
+        $apiClient->setOptions($this->container->getParameter('swp_bridge.options'));
+        return $apiClient;
```
Comment (ahilles107) [inline Controller/BridgeController.php]: missing empty line before return

File: README.md [hunk 2]
```diff
@@ -52,17 +52,40 @@ swp_bridge:
 
 ### Step 4: Configure the SWPBridgeBundle
 
-Add the following parameter to your `parameters.yml` file.
+Add the following parameters to your `parameters.yml` file.
 
 ```yaml
 # app/config/config.yml
```
Comment (ahilles107) [inline README.md]: config.yml or parameters.yml ?

Comment (m038) [inline README.md]: We add everything to parameters.yml, that gets loaded in config.yml afaik.

Comment (ahilles107) [inline README.md]: I know but this line misleads users.

Comment (m038) [inline README.md]: Why? They should add this into parameters.yml.

Comment (ahilles107) [inline README.md]: Yeah, but in example you have 
```
# app/config/config.yml
```

Comment (m038) [inline README.md]: Sorry, i wasn't paying attention :) Fixed

File: Client/GuzzleApiClient.php [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    protected $options = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function makeApiCall(RequestInterface $request)
+    {
+        $response = null;
+
+        if ($this->authenticator->getAccessToken() !== null) {
```
Comment (takeit) [inline Client/GuzzleApiClient.php]: I would move this if statement to private method instead. The code will be more readable and the method wont be so long

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -34,9 +34,42 @@ public function getConfigTreeBuilder()
 
         $rootNode
             ->children()
-                ->scalarNode('base_uri')
-                    ->cannotBeEmpty()
-                    ->isRequired()
+                ->arrayNode('api')
+                    ->children()
+                        ->scalarNode('host')
+                            ->info('Hostname where the Content API.')
```
Comment (takeit) [inline DependencyInjection/Configuration.php]: shouldnt be `Hostname of the Content API.` ?

File: DependencyInjection/SWPBridgeExtension.php [hunk 5]
```diff
@@ -31,16 +31,37 @@ class SWPBridgeExtension extends Extension
...
     public function load(array $configs, ContainerBuilder $container)
     {
+        $defaultOptions = array();
         $configuration = new Configuration();
         $config = $this->processConfiguration($configuration, $configs);
 
-        if (!empty($config['base_uri'])) {
-            $container->setParameter($this->getAlias().'.base_uri', $config['base_uri']);
+        if (isset($config['api'])) {
+            if (!empty($config['api']['host'])) {
```
Comment (takeit) [inline DependencyInjection/SWPBridgeExtension.php]: you can repalce those duplicated if statements with foreach loop. Something like:

```php
$keys = array('host', 'port', 'protocol');
foreach($keys as $value) {
    if (!empty($config['api'][$value])) {
         $container->setParameter($this->getAlias() . '.api.'.$value, $config['api'][$value]);
    }
} 
```

File: DependencyInjection/SWPBridgeExtension.php [hunk 6]
```diff
@@ -31,16 +31,37 @@ class SWPBridgeExtension extends Extension
...
+                $container->setParameter($this->getAlias() . '.api.protocol', $config['api']['protocol']);
+            }
         }
-        if (!empty($config['options'])) {
-            $container->setParameter($this->getAlias().'.options', $config['options']);
-        } else {
-            $container->setParameter($this->getAlias().'.options', array());
+
+        if (isset($config['auth'])) {
+            if (!empty($config['auth']['client_id'])) {
```
Comment (takeit) [inline DependencyInjection/SWPBridgeExtension.php]: same here. See the above comment

File: README.md [hunk 7]
```diff
@@ -52,17 +52,40 @@ swp_bridge:
 
 ### Step 4: Configure the SWPBridgeBundle
 
-Add the following parameter to your `parameters.yml` file.
+Add the following parameters to your `parameters.yml` file.
```
Comment (takeit) [inline README.md]: why into `parameters.yml` file? the config should be placed into `config.yml` file. Look at your `Configuration.php` class which is responsible for validating and merging config files not `parameters.yml`

File: Client/GuzzleApiClient.php (hunk 3 above)
Comment (m038) [inline Client/GuzzleApiClient.php]: @takeit please see https://github.com/superdesk/contentapi-sdk-php/pull/19 for an optimized version. I decided to move the generic code to the SDK, this class will just become an extension of DefaultApiClient.

**Comment by ahilles107** [comment]
tests are failing - https://travis-ci.org/SuperdeskWebPublisher/SWPBridgeBundle/jobs/102073300

**Comment by m038** [comment]
Looks like composer cache to me on travis... latest version of the SDK has this Class for sure.

**Pull request merged**